packages/runtime/src/widgets/DataGrid/DataCell.tsx

```diff
@@ -10,14 +15,32 @@ export interface DataCellProps {
   columnIndex: number;
 }
 export const DataCell: React.FC<DataCellProps> = ({
-  data,
   template,
   columnIndex,
+  data,
 }) => {
-  const cellTemplate = template.properties.children[columnIndex];
-  return (
-    <CustomScopeProvider value={data as CustomScope}>
-      {EnsembleRuntime.render([cellTemplate])}
-    </CustomScopeProvider>
-  );
+  const { "item-template": itemTemplate, children } = template.properties;
+  const { namedData } = useTemplateData({
+    ...itemTemplate,
+  });
+
+  const cellData = useMemo(() => {
```

Comment:
Let's get rid of this useMemo, I'm not sure there's a point here?

Reply:
There are two cases, either user pass the direct children array or the item template. and if user pass the item template then first we need to evaluate it's data expression into valid data object. and that's why I used here the memo.